pandas/tests/frame/test_nonunique_indexes.py

```diff
@@ -513,3 +513,23 @@ def test_set_value_by_index(self):
 
         df.iloc[:, 0] = 3
         tm.assert_series_equal(df.iloc[:, 1], expected)
+
+
+def test_iloc_with_boolean_operation():
+    # GH 20627
+    result = DataFrame([[0, 1], [2, 3], [4, 5], [6, np.nan]])
+    result.iloc[result.index <= 2] *= 2
```

Comment:
Could you also include a test where `iloc` also selects on the columns?

e.g. `result.iloc[result.index <= 2, 0] *= 2`

Reply:
ok done.